plugin/src/main/groovy/com/novoda/staticanalysis/internal/CodeQualityConfigurator.groovy

```diff
@@ -78,17 +111,29 @@ abstract class CodeQualityConfigurator<T extends SourceTask, E extends CodeQuali
         }
     }
 
-    protected abstract void configureAndroidVariant(variant)
-
-    protected void configureJavaProject() {
-        project.tasks.withType(taskClass) { task ->
-            sourceFilter.applyTo(task)
-            task.exclude '**/*.kt'
+    protected static def javaCompile(variant) {
+        if (variant.hasProperty('javaCompileProvider')) {
+            variant.javaCompileProvider.get()
+        } else {
+            variant.javaCompile
         }
     }
 
+    protected final String getToolTaskNameFor(sourceSet) {
```

Comment:
_technically_ this isn't always a sourceset right? maybe we can just called it `named`?

Reply:
Soooo, this was actually only for `sourceSet`. Then I realized that `FindBugs` needs to be configured per variant because it needs `javaCompile` task. I will go over and make sure that naming is consistent.